xbmc/storage/DetectDVDType.cpp

```diff
@@ -108,84 +100,85 @@ void CDetectDVDMedia::UpdateDvdrom()
   // newly inserted media.
   {
     std::unique_lock<CCriticalSection> waitLock(m_muReadingMedia);
-    switch (GetTrayState())
+    switch (PollDriveState())
     {
-      case DRIVE_NONE:
+      case DriveState::DRIVE_NONE:
         //! @todo reduce / stop polling for drive updates
         break;
 
-      case DRIVE_OPEN:
+      case DriveState::DRIVE_OPEN:
+      {
+        // Send Message to GUI that disc been ejected
+        SetNewDVDShareUrl(CServiceBroker::GetMediaManager().TranslateDevicePath(m_diskPath), false,
+                          g_localizeStrings.Get(502));
+        CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_REMOVED_MEDIA);
+        CServiceBroker::GetGUI()->GetWindowManager().SendThreadMessage(msg);
+        // Clear all stored info
+        Clear();
+        // Update drive state
+        waitLock.unlock();
+        m_DriveState = DriveState::DRIVE_OPEN;
+        return;
+      }
+      break;
+      case DriveState::DRIVE_NOT_READY:
+      {
+        // Drive is not ready (closing, opening)
+        SetNewDVDShareUrl(CServiceBroker::GetMediaManager().TranslateDevicePath(m_diskPath), false,
+                          g_localizeStrings.Get(503));
+        m_DriveState = DriveState::DRIVE_NOT_READY;
+        // DVD-ROM in undefined state
+        // Better delete old CD Information
+        if (m_pCdInfo != nullptr)
         {
-          // Send Message to GUI that disc been ejected
-          SetNewDVDShareUrl(CServiceBroker::GetMediaManager().TranslateDevicePath(m_diskPath),
-                            false, g_localizeStrings.Get(502));
-          CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_REMOVED_MEDIA);
-          CServiceBroker::GetGUI()->GetWindowManager().SendThreadMessage( msg );
-          // Clear all stored info
-          Clear();
-          // Update drive state
-          waitLock.unlock();
-          m_DriveState = DRIVE_OPEN;
-          return;
+          delete m_pCdInfo;
+          m_pCdInfo = nullptr;
```

Comment:
looking at this, could m_pCdInfo be a `std::unique_ptr`?

Reply:
for sure, and wrapped by another class as the cdinfo is actually a handler for cd based stuff. Not to be changed now